_rules/invalid-form-field-value-36b590.md

```diff
@@ -0,0 +1,203 @@
+---
+id: 36b590
+name: Error message describes invalid form field value
+rule_type: atomic
+description: |
+  This rule checks that text descriptions are provided when the user completes a form field with information that is not an allowed value or using a not allowed format.
+accessibility_requirements:
+  wcag20:3.3.1: # Error Identification (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag-technique:G84: # Providing a text description when the user provides information that is not in the list of allowed values
+    forConformance: false
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+  wcag-technique:G85: # Providing a text description when user input falls outside the required format or values
+    forConformance: false
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Language
+acknowledgements:
+  authors:
+    - Carlos Duarte
+    - João Vicente
+---
+
+## Applicability
+
+The rule applies to each HTML [input element][] with the [type attribute][] in one of the following [states][]: "Text", "Telephone", "URL", "E-mail", "Password", "Date", "Month", "Week", "Time", "Local Date and Time", and "Number";
+
+**Note**: The list of applicable [states][] includes those where the [input element][] can be rendered by the user agent as a text control.
+
+## Expectation 1
+
+For each test target the user [interacted with][], either after the interaction or after triggering the submission of a form if the target element belongs to one, a message is presented to the user for each [automatically detected error][].
+
+## Expectation 2
+
+The content of the message is [visible][], [included in the accessibility tree][], and identifies the [automatically detected error][].
```

Comment:
Isn't stating the error message is visible and included in the accessibility tree enough?

Reply:
> Only nodes can be included in the accessibility tree. But someone could also make the error message available through aria-errormessage, or through aria-label(ledby) / aria-describedby. I think you should explicitly allow those options.

`aria-label` would only work for AT users, and not help users with cognitive disabilities that don't use AT (`aria-errormessage`/`aria-labelledby`/`aria-describedby` point to another element, so assuming that THAT element is visible and in the a11y tree, so that's not AT specific)